CMakeLists.txt

```diff
@@ -114,7 +121,7 @@ endif()
 # Use libtcmalloc for Debug builds memory leaks detection
 set(DBG_FLAGS "-ggdb -gdwarf-2 -O0  ${WARN_FLAGS} -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free")
 
-if (CMAKE_BUILD_TYPE STREQUAL "Release")
+if (NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
```

Comment:
What's the intent behind this particular change? Are new build type being introduced? If so, could you point to where that'd be as I didn't see any changes here. (_Compiler ID is not build type, correct?_)

Reply:
CMAKE has several build types, such as RelWithDebInfo. See [CMAKE_BUILD_TYPE](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html) for more info. this change is meant to allow them.

Reply:
Just for clarification, Debug behaves as it did today, but all other flavors are now considered `Release` equivalent. Did I understand that correctly?

Reply:
Correct